src/sqlfluff/core/errors.py

```diff
@@ -47,11 +47,13 @@ def __init__(
             self.line_pos = line_pos
         super().__init__(self.desc())
 
-    def __reduce__(self):  # pragma: no cover
+    def __reduce__(
+        self,
+    ) -> Tuple[Type["SQLBaseError"], Tuple[Any, ...]]:  # pragma: no cover
         """Prepare the SQLBaseError for pickling."""
         return type(self), (
             self.description,
-            self.pos,
+            None,
```

Comment:
This is a positional arg, does anything tnat's using this expect something for it?

Reply:
I don't think so? This method is just for pickling, and the `pos` is just used creating for the initial instance., after which is splits that into line number and position. 

We also type the `pos` arg as `Optional` so that means it should be good to be `None`